### DISCUSSION TITLE
Ignore files from `node ace serve --watch`

### DISCUSSION THREAD

**Discussion by jacobq** [adonisjs/core#1418] [Help] [answered]
I have a task running in the background that generates some plain files like `build/public/folder-to-ignore/foo/bar.json` that I want my AdonisJS app to serve as static files. This works fine, but I can't simultaneously use the `--watch` (default `npm start`) feature or it will get stuck in an infinite loop. Is there some way that I can specify a glob/pattern or folder to ignore?

**Comment by iammrsea** [comment, marked as answer]
How often does your background task that generates the plain files run? If it's frequent, that could explain why you are experiencing an infinite loop behavior because with **--watch** flag, your server  restarts any time **public** folder changes. I assume you registered your public folder in **.adonisrc.json**, and if that's the case, you can modify it like this:
```javascript
{
 "metaFiles":[
      {
        "pattern": "public/**",
        "reloadServer: false
     }
    ]
}
```

**Comment by thetutlage** [comment]
As you are writing files directly to the `build` folder. I don't think, the AdonisJS watcher will cause any issues, since it ignores the `build` folder by default

**Reply by jacobq** [reply]
@iammrsea @thetutlage Thanks for your replies. The files were getting generated on start-up and then about every 5 minutes. Looks like I was writing to `public` (though perhaps to `build/public` in others if launched some other way than `npm start` / `ace serve --watch`). Admittedly that's not a great setup, but it's working for now. The `reloadServer` option did the trick (though the code above is missing a `"`). I was also using `__dirname` as part of a default path; now that I'm using an env var for this I don't have to second guess.

**Comment by ivan-suhorukov** [comment]
Hello! I'm trying to combine adonisjs with the vite on frontend and i put my `frontend` folder in root folder of adonis. Next i'm runing `vite serve` and `node ace serve --watch` in parallel. It works, but now ace server restarts every time i get change in my frontend folder. I don't want to add my frontend folder in `metaFiles` section of config, because i don't need this folder in build output. So my question is the same: can i exclude folder from ace watching? Or maybe there is better way to do all of this?
